### PR TITLE
Close Blank Tabs on App Foreground

### PR DESCRIPTION
**Pull request by evancrow**
Whenever the app is backgrounded, the app will now quietly close tabs whose URL is `about:blank`.

### How was this tested?
- [x] I tested this manually

### Screenshots and screen recordings

https://user-images.githubusercontent.com/37356984/172252355-d792b762-d35e-4744-bdef-ce5751b3ff95.MP4

## Issues addressed
Fixed: #3766

### CODE REVIEW

File: Client/Frontend/Browser/Tabs/TabManager/TabManager+CloseTabs.swift [hunk 1]
```diff
@@ -190,4 +190,10 @@ extension TabManager {
             }
         }
     }
+
+    // MARK: - Blank Tabs
+    /// Removes and closes blank tabs opened from `target=_blank` or `about:blank`.
```
Comment (jedireza) [inline Client/Frontend/Browser/Tabs/TabManager/TabManager+CloseTabs.swift:195]:
```suggestion
    /// Removes any tabs with the location `about:blank`. Seen when clicking web links that open native apps.
```

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -811,13 +811,6 @@ class BrowserViewController: UIViewController, ModalPresenter {
         )
     }
 
-    func openBlankNewTab(isIncognito: Bool = false) {
```
Comment (jedireza) [inline Client/Frontend/Browser/BrowserViewController.swift:814]: Found some dead code I presume.

**Review by jedireza — APPROVED** [review]
Nice to see this approach worked out. Code looks good. I left a suggestion for the comment line; your call. 🚀

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (evancrow) [reply Client/Frontend/Browser/BrowserViewController.swift:814]: That’s right!

**Review by EdwardJTL — COMMENTED** [review]
as a thought: how do we feel about doing the clean up on app enter foreground? or before making the system call to open an external app?
bringing this up because I was under the impression that we have limited time/resources to save tabs when app is going into background

**Comment by evancrow** [comment]
Interesting thought. I'm not sure how much of an impact this will have, but I think that making the move to willEnterForeground could be better. Will try it out and report back!

**Comment by evancrow** [comment]
@EdwardJTL Moved it to `willEnterForeground`. Tried it with 600+ tabs and performance is good!

**Pull request merged**